forms-flow-bpm/forms-flow-bpm-camunda/src/main/java/org/camunda/bpm/extension/hooks/listeners/task/NotifyListener.java

```diff
@@ -46,74 +45,85 @@ public class NotifyListener extends BaseListener implements TaskListener, IMessa
      * @param delegateTask: The task which sends the message
      */
     public void notify(DelegateTask delegateTask) {
-        Set<String> toEmails =  new HashSet<>();
-        if(!"Y".equals(getGroupsOnly(delegateTask.getExecution()))) {
+        Set<String> toEmails = new HashSet<>();
+        if (!"Y".equals(getGroupsOnly(delegateTask.getExecution()))) {
             toEmails.addAll(getEmailsOfUnassignedTask(delegateTask));
         }
         List<String> emailGroups = null;
         try {
             emailGroups = getEmailGroups(delegateTask.getExecution());
         } catch (IOException e) {
-           handleException(delegateTask.getExecution(), ExceptionSource.TASK, e);
+            handleException(delegateTask.getExecution(), ExceptionSource.TASK, e);
         }
         if (CollectionUtils.isNotEmpty(emailGroups)) {
             for (String entry : emailGroups) {
                 toEmails.addAll(getEmailsForGroup(delegateTask.getExecution(), entry));
             }
         }
-        sendEmailNotification(delegateTask, toEmails ,delegateTask.getId());
+        try {
+            sendEmailNotification(delegateTask, toEmails, delegateTask.getId());
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
     }
 
     /**
-     *
      * @param delegateTask
      * @param toEmails
      * @param taskId
      */
-    private void sendEmailNotification(DelegateTask delegateTask,Set<String> toEmails,String taskId) {
-        String toAddress = CollectionUtils.isNotEmpty(toEmails) ? StringUtils.join(toEmails,",") : null;
-        if(StringUtils.isNotEmpty(toAddress)) {
+    private void sendEmailNotification(DelegateTask delegateTask, Set<String> toEmails, String taskId) throws JsonProcessingException {
+        toEmails.addAll(getEmailAddress(delegateTask.getExecution()));
+        String toAddress = CollectionUtils.isNotEmpty(toEmails) ? StringUtils.join(toEmails, ",") : null;
+        if (StringUtils.isNotEmpty(toAddress)) {
             Map<String, Object> emailAttributes = new HashMap<>();
             emailAttributes.put(EMAIL_TO, toAddress);
             emailAttributes.put("category", getCategory(delegateTask.getExecution()));
-            emailAttributes.put("name",getDefaultAddresseName());
-            emailAttributes.put("taskid",taskId);
-            if(StringUtils.isNotBlank(toAddress) && StringUtils.indexOf(toAddress,"@") > 0) {
-                sendMessage(delegateTask.getExecution(), emailAttributes,getMessageId(delegateTask.getExecution()));
+            emailAttributes.put("name", getDefaultAddresseName());
```

Comment:
Can we send only these attributes via email, or is it possible to attach any body or files as well?

Reply:
Actually it is the internal logic of notify listener to send the mail notification. Email Configuration data such as body, subject attachments are handled by `EmailAttributeListener` which is configured in our internal mail workflow. By adding this notify listener it will invoke the internal workflow and send the mail using 1EmailAttributeListener`